scipy/optimize/_linprog.py

```diff
@@ -357,7 +357,10 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     Returns
     -------
     res : OptimizeResult
-        A :class:`scipy.optimize.OptimizeResult` consisting of the fields:
+        A :class:`scipy.optimize.OptimizeResult` consisting of the fields
+        below. Note that the return types of the fields may depend on whether
+        the optimization was successful, therefore it is recommended to check
+        `OptimizeResult.success` before relying on the other fields:
```

Comment:
```suggestion
        `OptimizeResult.status` before relying on the other fields:
```
What do you think of recommending `status` instead of `success`, since it has more information?

Reply:
Fine by me! I thought `success` made sense since it's a boolean where `True` corresponds to `OptimizeResult`'s attributes having the expected types, but `False` means a bunch of the attributes may be `None`. Both `success` and `status` are equally easy to check programmatically though, so I don't have a strong opinion.